File: Reservationpitch.Application/DTOs/StadiumCenterDtos/Request/StadiumCenterRequestDto.cs
using Microsoft.AspNetCore.Http;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Reservationpitch.Application.DTOs.StadiumDTOs.Request
{
    public class StadiumCenterRequestDto
    {
        [Required]
        public string Name { get; set; }
        public string PhoneNumber { get; set; }
        public string Location { get; set; }
        public string Owned_By { get; set; } = string.Empty;
        public bool DressingRoomVisible { get; set; } = false;
        public string ImageUrl { get; set; }
        public IFormFile file {  get; set; }
    }
}

File: App/stadium_reservation/lib/models/Dtos/stadiumcenterrequestdto.dart
class Stadiumcenterrequestdto {
  String name;
  String phoneNumber;
  String location;
  String owned_By;
  bool dressingRoomVisible;

  Stadiumcenterrequestdto({
    required this.name,
    required this.phoneNumber,
    required this.location,
    required this.owned_By,
    required this.dressingRoomVisible,
  });

  // Convert the DTO to a JSON object to send in the POST request
  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'phoneNumber': phoneNumber,
      'location': location,
      'owned_By': owned_By,
      'dressingRoomVisible': dressingRoomVisible,
    };
  }
}
